src/mediastream.h

```diff
@@ -0,0 +1,67 @@
+/* Copyright (c) 2018 The node-webrtc project authors. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-style license that can be found
+ * in the LICENSE.md file in the root of the source tree. All contributing
+ * project authors may be found in the AUTHORS file in the root of the source
+ * tree.
+ */
+#ifndef SRC_MEDIASTREAM_H_
+#define SRC_MEDIASTREAM_H_
+
+#include "nan.h"
+#include "v8.h"
+
+#include "webrtc/api/mediastream.h"
+
+#include "src/mediastreamtrack.h"
+#include "src/peerconnectionfactory.h"
+
+namespace node_webrtc {
+
+class MediaStream
+  : public Nan::AsyncResource
+  , public Nan::ObjectWrap {
+ public:
+  MediaStream(
+      std::shared_ptr<node_webrtc::PeerConnectionFactory>&& factory,
+      rtc::scoped_refptr<webrtc::MediaStreamInterface>&& stream):
+    MediaStream(std::move(factory), std::move(stream), !_shouldReleaseFactory) {}
+
+  MediaStream(
+      std::shared_ptr<node_webrtc::PeerConnectionFactory>&& factory,
+      rtc::scoped_refptr<webrtc::MediaStreamInterface>&& stream,
+      bool shouldReleaseFactory);
```

Comment:
Would be good to comment on these. The reason you might pass a non-null `factory` _and_ set `shouldReleaseFactory` to `true` are a result of the `new MediaStream(tracks)` constructor and needing to use the default factory. Actually, maybe we could still encapsulate this logic inside the "smaller" constructor when `factory` and `stream` are `nullptr`.